thrust/system/cuda/detail/bulk/malloc.hpp

```diff
@@ -38,9 +38,24 @@ inline __device__ bool is_on_chip(void *ptr)
 template<typename T>
 inline __device__ T *on_chip_cast(T *ptr)
 {
+#if defined(__CUDA__) && defined(__NVCC__) && !defined(__clang__)
```

Comment:
This looks good to me and I'm okay with this ifdef for now. I am not sure what you suggest at the bottom though, but would it make sense for clang to define __clang_cuda__ macro, or alike; or is this something better be done in Thrust?